dev/io.openliberty.org.jboss.resteasy.mprestclient/src/org/jboss/resteasy/microprofile/client/ot/LibertyRestClientBuilderImpl.java

```diff
@@ -116,12 +127,30 @@
     public static final MethodInjectionFilter METHOD_INJECTION_FILTER = new MethodInjectionFilter();
     public static final ClientHeadersRequestFilter HEADERS_REQUEST_FILTER = new ClientHeadersRequestFilter();
 
+    private static final Class<?> FT_ANNO_CLASS = getFTAnnotationClass();
+
     static ResteasyProviderFactory PROVIDER_FACTORY;
 
     public static void setProviderFactory(ResteasyProviderFactory providerFactory) {
         PROVIDER_FACTORY = providerFactory;
     }
 
+    @FFDCIgnore(PrivilegedActionException.class)
+    private static Class<?> getFTAnnotationClass() {
+        try {
+            return AccessController.doPrivileged(
+                (PrivilegedExceptionAction<Class<?>>) () -> {
+                    return Class.forName("com.ibm.ws.microprofile.faulttolerance.cdi.FaultTolerance");
+            });
+        } catch (PrivilegedActionException pae) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Exception checking for MP Fault Tolerance class - " +
+                             "expected if FT feature isnot enabled", pae);
+            }
+            return null;
+        }
+    }
+
```

Comment:
We should probably have a proper integration point rather than just looking to see if an internal class is loadable.

I don't understand enough about what else is going on in this class to make a sensible suggestion about what it should be though. Could we have a chat about the best way to do this?